src/OperationResponse.php

```diff
@@ -0,0 +1,230 @@
+     */
+    public function pollUntilComplete($options = [])
+    {
+        $defaultPollSettings = [
+            'pollingIntervalSeconds' => $this::DEFAULT_POLLING_INTERVAL,
+        ];
+        $pollSettings = array_merge($defaultPollSettings, $options);
+
+        $pollingIntervalMicros = $pollSettings['pollingIntervalSeconds'] * 1000000;
+
+        while (!$this->isDone()) {
+            usleep($pollingIntervalMicros);
+            $this->reload();
+        }
+
+        return $this->lastProtoResponse->hasResponse();
+    }
+
+    /**
+     * Reload the status of the operation with a request to the service.
+     */
+    public function reload()
+    {
+        if ($this->deleted) {
+            throw new ValidationException("Cannot call reload() on a deleted operation");
+        }
+        $name = $this->getName();
+        $this->lastProtoResponse = $this->operationsClient->getOperation($name);
+    }
+
+    /**
+     * Return the result of the operation. If the operation is not complete, or if the operation
+     * failed, return null.
+     * @return mixed The result of the operation, or null if the operation failed or is not complete
+     */
+    public function getResult()
+    {
+        if (!$this->isDone() || !$this->lastProtoResponse->hasResponse()) {
+            return null;
+        }
+
+        $anyResponse = $this->lastProtoResponse->getResponse();
+        if (is_null($this->operationReturnType)) {
+            return $anyResponse;
+        }
+        $operationReturnType = $this->operationReturnType;
+        $response = new $operationReturnType();
+        $response->parse($anyResponse->getValue());
+        return $response;
+    }
+
+    /**
+     * If the operation failed, return the status. If the operation succeeded or is not complete,
+     * return null.
+     *
+     * @return \google\rpc\Status|null The status of the operation in case of failure, otherwise
+     *                                 null.
+     */
+    public function getError()
+    {
+        if (!$this->isDone() || !$this->lastProtoResponse->hasError()) {
+            return null;
+        }
+        return $this->lastProtoResponse->getError();
+    }
+
+    /**
+     * @return \google\longrunning\Operation The last Operation object received from the server.
+     */
+    public function getLastProtoResponse()
+    {
+        return $this->lastProtoResponse;
+    }
+
+    /**
+     * @return \Google\Longrunning\OperationsClient The OperationsClient object used to make
+     * requests to the operations API.
+     */
+    public function getOperationsClient()
+    {
+        return $this->operationsClient;
+    }
+
+    /**
+     * Starts asynchronous cancellation on a long-running operation. The server
+     * makes a best effort to cancel the operation, but success is not
+     * guaranteed. If the server doesn't support this method, it will throw an
+     * ApiException with code \google\rpc\Code::UNIMPLEMENTED. Clients can continue
```

Comment:
Since we are documenting in the description, would it make sense to also add an `@throws` tag?